src/main/java/de/bonndan/nivio/output/map/hex/GroupAreaFactory.java

```diff
@@ -0,0 +1,101 @@
+package de.bonndan.nivio.output.map.hex;
+
+import de.bonndan.nivio.model.Group;
+import de.bonndan.nivio.model.Item;
+import de.bonndan.nivio.model.LandscapeItem;
+import de.bonndan.nivio.output.map.svg.HexPath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Collects all hexes close to group item hexes to create an area.
+ */
+public class GroupAreaFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GroupAreaFactory.class);
+
+    /**
+     * Builds an areas of hex tiles belonging to a group.
+     *  @param occupied    tiles occupied by items
+     * @param group       the group
+     * @param vertexHexes a mapping from item to its hex
+     * @param relationPaths   existing paths
+     * @return
+     */
+    public static Set<Hex> getGroup(Set<Hex> occupied, Group group, Map<LandscapeItem, Hex> vertexHexes, List<HexPath> relationPaths) {
+
+        Set<Item> items = group.getItems();
+        Set<Hex> inArea = new HashSet<>();
+
+        //surround each item
+        items.forEach(item -> {
+            Hex hex = vertexHexes.get(item);
+            inArea.add(hex);
+            hex.neighbours().forEach(neigh -> {
+                if (!occupied.contains(neigh))
+                    inArea.add(neigh);
+            });
+
+            //add all "inner" relations (paths)
+            relationPaths.stream()
+                    .filter(rel -> rel.getSource().equals(item))
+                    .filter(rel -> rel.getTarget().getGroup() != null)
+                    .filter(rel -> rel.getTarget().getGroup().equals(group.getIdentifier()))
+                    .forEach(rel -> inArea.addAll(rel.getHexes()));
+        });
+
+        Set<Hex> bridges = getBridges(inArea);
+        inArea.addAll(bridges);
+
+        //2nd pass fills gaps
+        bridges = getBridges(inArea);
+        inArea.addAll(bridges);
```

Comment:
It works as follows: first we circumnavigate all hexes of items and add their neighbours immediately. Then we iterate over all one-hex gaps and add them. This iteration is repeated, so that effectively a few two-hex gaps were filled.

There is clearly much room for improvement here. It's only that I haven't found a bettter approach so far.

Reply:
That explanation would be a greate code comment.